src/casp/services/api/exception_handlers.py

```diff
@@ -1,33 +1,66 @@
-from fastapi import Request, responses
+from fastapi import Request, responses, status
 
-from casp.services.api.services.exceptions import AccessDeniedError, InvalidInputError
+from casp.services._auth.exceptions import TokenException
+from casp.services.api.data_manager.exceptions import NotFound
+from casp.services.api.services.exceptions import APIBaseException
 
 
-async def access_denied_error_handler(_: Request, exc: AccessDeniedError) -> responses.JSONResponse:
+async def api_base_exception_handler(_: Request, exc: APIBaseException) -> responses.JSONResponse:
     """
-    Handler for AccessDeniedError exceptions
+    Handler for APIBaseException which sets the code from the exception type.
 
     :param _:
-    :param exc: AccessDeniedError exception
+    :param exc: APIBaseException exception
 
-    :return: Response with 403 status code and error message
+    :return: Response with status code from the exception and its error message.
     """
     return responses.JSONResponse(
-        status_code=403,
+        status_code=exc.http_code,
         content={"detail": str(exc)},
     )
 
 
-async def invalid_input_error_handler(_: Request, exc: InvalidInputError) -> responses.JSONResponse:
+async def not_found_error_handler(_: Request, exc: NotFound) -> responses.JSONResponse:
     """
-    Handler for InvalidInputError exceptions
+    Handler for NotFound exceptions
 
     :param _:
-    :param exc: InvalidInputError exception
+    :param exc: NotFound exception
 
     :return: Response with 400 status code and error message
     """
     return responses.JSONResponse(
-        status_code=400,
+        status_code=404,
         content={"detail": str(exc)},
     )
+
+
+async def token_exception_handler(_: Request, exc: TokenException) -> responses.JSONResponse:
+    """
+    Handler for TokenException which sets the code from the exception type.
+
+    :param _:
+    :param exc: TokenException exception
+
+    :return: Response with status code from the exception and its error message.
+    """
+    return responses.JSONResponse(
+        status_code=exc.http_code,
+        content={"detail": str(exc)},
+        headers={"WWW-Authenticate": "Bearer"},
+    )
+
+
+async def global_error_handler(*_: object) -> responses.JSONResponse:
```

Comment:
```suggestion
async def global_error_handler(*args, **kwargs) -> responses.JSONResponse:
```

Reply:
I wasn't sure about that.  This way (`*_`) I don't get the args not access warnings.  OK if I leave like this?